tests/test_buku.py

```diff
@@ -997,3 +1003,28 @@ def test_parse_range(tokens, valid, expected):
         except Exception as e:
             assert type(e) is type(expected)
             assert str(e) == str(expected)
+
+
+def test_split_by_marker():
+    search_string = (' global substring  .title substring :url substring :https '
+                     '> description substring #partial,tags: #,exact,tags, *another global substring ')
+    assert split_by_marker(search_string) == [
+        ' global substring', '.title substring', ':url substring', ':https',
+        '> description substring', '#partial,tags:', '#,exact,tags,', '*another global substring ',
+    ]
+
+
+def test_SortKey():
+    assert repr(SortKey('foo', ascending=True)) == "+'foo'"
+    assert repr(SortKey('bar', ascending=False)) == "-'bar'"
+    assert SortKey('foo', ascending=True) > SortKey('bar', ascending=True)
+    assert not SortKey('foo', ascending=True) > SortKey('foo', ascending=True)  # pylint: disable=unnecessary-negation
+    assert not SortKey('foo', ascending=True) < SortKey('foo', ascending=True)  # pylint: disable=unnecessary-negation
+    assert not SortKey('foo', ascending=True) < SortKey('bar', ascending=True)  # pylint: disable=unnecessary-negation
+    assert SortKey('foo', ascending=False) < SortKey('bar', ascending=False)
+    assert not SortKey('foo', ascending=False) < SortKey('foo', ascending=False)  # pylint: disable=unnecessary-negation
+    assert not SortKey('foo', ascending=False) > SortKey('foo', ascending=False)  # pylint: disable=unnecessary-negation
+    assert not SortKey('foo', ascending=False) > SortKey('bar', ascending=False)  # pylint: disable=unnecessary-negation
+
+    custom_order = lambda s: (SortKey(len(s), ascending=False), SortKey(s, ascending=True))
+    assert sorted(['foo', 'bar', 'baz', 'quux'], key=custom_order) == ['quux', 'bar', 'baz', 'foo']
```

Comment:
“Sort longer strings first, ordering lexicographically when they have the same length”